src/cascade_at/executor/args/args.py

```diff
@@ -10,93 +10,109 @@ class StaticArgError(CascadeArgError):
 
 
 class _Argument:
+    """
+    Base class for all arguments. By default, all arguments
+    are considered node_args, which is something that makes
+    the command that it is used in unique within a template
+    for the workflow, but not across workflows.
+
+    This is overwritten
+    if an _Argument subclass is used as a task_arg.
+    """
     def __init__(self, arg=None):
 
-        self._arg = None
-        self._kwargs = dict()
+        self._flag = None
+        self._parser_kwargs = dict()
         if arg is not None:
-            self._arg = arg
+            self._flag = arg
+        self._task_arg = False
```

Comment:
So all args are node_args by default? Is this intentional or am I missing somewhere task_args are set later?

Reply:
Yep! They're all `node_args`. The only thing that defines data flow is `model_version_id` so that's the only `task_arg`